### DISCUSSION TITLE
Data migrations

### DISCUSSION THREAD

**Discussion by etmartinkazoo** [payloadcms/payload#625] [Feature Requests & Ideas]
We just completed a web project and are using Payload CMS in production; great tool!

We ran into an issue whereby improperly named fields - e.g. 'locking-system' - broke the server connection to GraphQL. Lesson learned, always use camel case or snake case. The error message was:

`Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but \"locking-system\" does not.`

While there seem to be two NPM packages for migrations with Mongo/Mongoose, some sort of database migration tool would be critical with a larger data set.

Does Payload have any plans in this regard or solution they recommend from previous project experience?

Thank you!

**Comment by DanRibbens** [comment]
Ah! This one slipped by unanswered, we've had a few discussions about what can be done to handle migrations as data changes. With no-sql you don't hit errors like you might with SQL but it can lead to nasty bugs for frontends that receive bad data.

In other [disscussions](https://github.com/payloadcms/payload/discussions/664#discussioncomment-2972384) we've recommended using the [mongoose-migrate](https://www.npmjs.com/package/migrate-mongoose) package.

Would love to hear your experience if you do go this route.

**Reply by etmartinkazoo** [reply]
We will use mongoose-migrate in the future and report back.

We build a lot of business applications and have used Rails or Django with PostgreSQL in the past. We are using Payload CMS for all projects going forward - we have four in production serving businesses at the time of writing this. The developer experience even prior to your 1.0 release has just been phenomenal. I think we would go back to PostgreSQL in the future if that were an option but on the whole we couldn't be happier with Payload CMS. Thank you for making the internet a better place!

**Reply by MattRvt** [reply]
mongoose-migrate does not seem to be maintained anymore. (https://github.com/balmasi/migrate-mongoose/issues/62)

[node-db-migrate](https://github.com/db-migrate/node-db-migrate) seem interesting, but I failed to configure it.

**Reply by DanRibbens** [reply]
There is this package also https://github.com/seppevs/migrate-mongo which has more recent updates. That could help until we have an ORM with migrations.

**Reply by MentalGear** [reply]
@etmartinkazoo Would you mind reporting back with your experiences data migrating and all payload so far?

**Comment by DanRibbens** [comment]
I made a sample repo that has migrations inside a Payload project using `migrate-mongo` and it worked pretty well in my testing, see https://github.com/payloadcms/migrate-mongo-example.

I'll write a blog post about it and depending on reception, it could be added to the Payload docs.

**Reply by charkour** [reply]
@DanRibbens, thanks for the example. Did you release that blog post?

**Reply by DanRibbens** [reply]
Not yet, thanks for asking. I really need to get after that!
